### PR TITLE
http server connection  https protocol support

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

http server connection  https protocol support

## Fixes

https://github.com/fetchai/agents-aea/issues/1650

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_packages/test_connections/test_http_server/test_http_server.py [hunk 1]
```diff
@@ -463,3 +462,104 @@ def test_bad_api_spec():
...
+        self.host = "localhost"
+        self.port = get_unused_tcp_port()
+        self.api_spec_path = os.path.join(
+            ROOT_DIR, "tests", "data", "petstore_sim.yaml"
+        )
+        self.connection_id = HTTPServerConnection.connection_id
+        self.protocol_id = HttpMessage.protocol_id
+        self.target_skill_id = "some_author/some_skill:0.1.0"
+        self.ssl_cert = os.path.join(ROOT_DIR, "tests", "data", "certs", "server.crt")
+        self.ssl_key = os.path.join(ROOT_DIR, "tests", "data", "certs", "server.key")
```
Comment (DavidMinarsch) [inline tests/test_packages/test_connections/test_http_server/test_http_server.py:506]: Did you forget to commit these files?

Comment (solarw) [reply tests/test_packages/test_connections/test_http_server/test_http_server.py:506]: right, thanls. added

**Review by DavidMinarsch — APPROVED** [review]
LGTM

File: packages/fetchai/connections/http_server/connection.py [hunk 2]
```diff
@@ -541,6 +557,12 @@ def __init__(self, **kwargs: Any) -> None:
         api_spec_path = cast(
             Optional[str], self.configuration.config.get("api_spec_path")
         )
+        ssl_cert_path = cast(Optional[str], self.configuration.config.get("ssl_cert"))
+        ssl_key_path = cast(Optional[str], self.configuration.config.get("ssl_key"))
+
+        if bool(ssl_cert_path) != bool(ssl_key_path):  # pragma: nocover
+            raise ValueError("Please specify both ssl_cert and ssl_key or none of it")
```
Comment (5A11) [inline packages/fetchai/connections/http_server/connection.py:564]:
```suggestion
            raise ValueError("Please specify both ssl_cert and ssl_key or neither.")
```

**Review by 5A11 — APPROVED** [review]

**Pull request merged**